docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+|                        |      | structure to names.                          |
+|------------------------|------|--------------------------------------------- |
+| getGivenName()         | N/A  | Not defined in the OpenYOLO specification,   |
+|                        |      | for the same reasons as for `getFamilyName`. |
+|-----|---------------------------|--------------------------------------------|
+| N/A | getAdditionalProperties() | OpenYOLO offers extensibility for Hints    |
+|     |                           | via a general map of strings to byte       |
+|     |                           | arrays.                                    |
+|-----|---------------------------|--------------------------------------------|
+
+
+## Saving Credentials
+
+When a credential is determined to be valid, it can be saved to the user's
+credential provider. In the Smart Lock for Passwords API, this was a two-step
+process similar to calling `retrieve()`:
+
+1. The credential is constructed and sent to Smart Lock for Passwords using the
+   `save()` method, with a callback provided. This callback is then invoked,
+   indicating whether the save succeeded, user consent is required to save
+   the credential (a `PendingIntent` is returned), or the save was rejected.
+
+2. If an Intent is returned, the developer must invoke this `PendingIntent` in
+   order to prompt the user to save the credential. The result of this is
+   indicated via `onActivityResult()`.
+
+To simplify the API in OpenYOLO, an Intent is directly created and used to
+save the credential.
+
+In Smart Lock for Passwords, a credential would be saved as follows:
+
+```java
+void saveSmartLockCredential(Credential credential) {
+  Auth.CredentialsApi.save(mGoogleApiClient, credential)
+      .setResultCallback(this::handleSmartLockSaveResult);
+}
+
+void handleSmartLockSaveResult(Status status) {
+  if (status.isSuccess()) {
+      // credential automatically saved
+  } else if (status.hasResolution()) {
+      try {
+          status.startResolutionForResult(this, RC_SAVE);
+      } catch (IntentSender.SendIntentException e) {
+          Log.e(TAG, "Oh no :(");
+      }
+  } else {
+      // save refused by provider or failed.
+      // ...
+  }
+}
+
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_SAVE) {
+    if (resultCode = RESULT_OK) {
+      // credential saved
+      // ...
+    } else {
+      // credential not saved
+      // ...
+    }
+  }
+}
+```
+
+With OpenYOLO, this is simplified to:
+
+```java
+void saveOpenYoloCredential(Credential credential) {
+  Intent saveIntent = mOpenYoloClient.getSaveIntent(credential);
+  startActivityForResult(saveIntent, RC_SAVE);
+}
+
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+  if (requestCode == RC_SAVE) {
+    if (resultCode == CredentialSaveResult.CODE_SAVED) {
```

Comment:
Same here, please use the resultObject.isSuccessful()

Reply:
Done.